### ISSUE TITLE
Old docker compose in WSL

### ISSUE THREAD

**Issue by CybotTM** [docker/for-win#12928]
- [X] I have tried with the latest version of Docker Desktop v4.11.1
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID:  ED9AAF6A-024C-40A0-8F34-3D7D1EF844A0/20220824095214

### Actual behavior
```
$ docker compose version
Docker Compose version v2.2.2
$ docker-compose version
Docker Compose version v2.2.2
```

### Expected behavior

```
$ docker compose version
Docker Compose version v2.7.0
$ docker-compose version
Docker Compose version v2.7.0
```

### Information

```
$ docker-compose version
Docker Compose version v2.2.2
$ uname -a
Linux Grosser 5.15.57.1-microsoft-standard-WSL2 #1 SMP Wed Jul 27 02:20:31 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
$ which docker-compose
/usr/bin/docker-compose
$ ll /usr/bin/docker-compose
lrwxrwxrwx 1 root root 56 Aug 24 09:03 /usr/bin/docker-compose -> /mnt/wsl/docker-desktop/cli-tools/usr/bin/docker-compose*
$ ll /mnt/wsl/docker-desktop/cli-tools/usr/bin/
total 154391
drwxr-xr-x 2 root root     2048 Jul 25 19:13 ./
drwxr-xr-x 6 root root     2048 Jul 25 19:13 ../
-rwxr-xr-x 1 root root 60513160 Jun  7 01:01 com.docker.cli*
-rwxr-xr-x 1 root root 55902208 Jul 25 19:12 docker*
-rwxr-xr-x 1 root root  6332416 Jul 25 19:13 docker-compose*
-rwxr-xr-x 1 root root 12737304 Jul 25 19:13 docker-compose-v1*
-rwxrwxr-x 1 root root  9401344 Jul 25 19:12 docker-credential-desktop.exe*
-rwxr-xr-x 1 root root 13205504 Jul 25 19:13 hub-tool*
```

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`

```
[2022-08-24T08:52:54.843379500Z][com.docker.diagnose.exe][I] set path configuration to OnHost
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0031: does the Docker API work?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0001: is the application running?
[SKIP] DD0018: does the host support virtualization?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0017: can a VM be started?
[PASS] DD0024: is WSL installed?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0035: is the VM time synchronized?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0005: is the user in the docker-users group?
[PASS] DD0007: is the backend responding?
[FAIL] DD0014: are the backend processes running? 1 error occurred:
        * vpnkit.exe is not running


[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0006: is the Docker Desktop Service responding?
[FAIL] DD0012: is the VM networking working? network checks failed: failed to ping host: exit status 1
[2022-08-24T08:53:02.554695400Z][com.docker.diagnose.exe][I] ipc.NewClient: e9cac12b-diagnose-network -> \\.\pipe\dockerDiagnosticd diagnosticsd
[common/pkg/diagkit/gather/diagnose.runIsVMNetworkingOK()
[       common/pkg/diagkit/gather/diagnose/network.go:34 +0xd9
[common/pkg/diagkit/gather/diagnose.(*test).GetResult(0x151cfe0)
[       common/pkg/diagkit/gather/diagnose/test.go:46 +0x43
[common/pkg/diagkit/gather/diagnose.Run.func1(0x151cfe0)
[       common/pkg/diagkit/gather/diagnose/run.go:17 +0x5a
[common/pkg/diagkit/gather/diagnose.walkOnce.func1(0x2?, 0x151cfe0)
[       common/pkg/diagkit/gather/diagnose/run.go:140 +0x77
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x1, 0x151cfe0, 0xc0002c7728)
[       common/pkg/diagkit/gather/diagnose/run.go:146 +0x36
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x0, 0xcb?, 0xc0002c7728)
[       common/pkg/diagkit/gather/diagnose/run.go:149 +0x73
[common/pkg/diagkit/gather/diagnose.walkOnce(0xed3a40?, 0xc0002df890)
[       common/pkg/diagkit/gather/diagnose/run.go:135 +0xcc
[common/pkg/diagkit/gather/diagnose.Run(0x151d5e0, 0x6a3c4b3100000010?, {0xc0002dfb20, 0x1, 0x1})
[       common/pkg/diagkit/gather/diagnose/run.go:16 +0x1ca
[main.checkCmd({0xc00007a3d0?, 0xc00007a3d0?, 0x4?}, {0x0, 0x0})
[       common/cmd/com.docker.diagnose/main.go:133 +0x105
[main.main()
[       common/cmd/com.docker.diagnose/main.go:99 +0x288
[2022-08-24T08:53:02.555198000Z][com.docker.diagnose.exe][I] (83638d07) e9cac12b-diagnose-network C->S diagnosticsd POST /check-network-connectivity: {"ips":["172.16.10.20","192.168.86.45","172.21.0.1","172.22.240.1"]}
[2022-08-24T08:53:03.091096500Z][com.docker.diagnose.exe][W] (83638d07) e9cac12b-diagnose-network C<-S 9f449bcf-diagnosticsd POST /check-network-connectivity (536.4011ms): failed to ping host: exit status 1

[FAIL] DD0032: do Docker networks overlap with host IPs? network server-playbooks_default has subnet 172.21.0.0/16 which overlaps with host IP 172.21.0.1
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?

Please investigate the following 3 issues:

1 : The test: are the backend processes running?
    Failed with: 1 error occurred:
        * vpnkit.exe is not running



Not all of the backend processes are running.

2 : The test: is the VM networking working?
    Failed with: network checks failed: failed to ping host: exit status 1

VM seems to have a network connectivity issue. Please check your host firewall and anti-virus settings in case they are blocking the VM.

3 : The test: do Docker networks overlap with host IPs?
    Failed with: network server-playbooks_default has subnet 172.21.0.0/16 which overlaps with host IP 172.21.0.1

If the subnet used by a Docker network overlaps with an IP used by the host, then containers
won't be able to contact the overlapping IP addresses.

Please try configuring the IP address range used by networks: in your docker-compose.yml.
See https://docs.docker.com/compose/compose-file/compose-file-v2/#ipv4_address-ipv6_address
```

### Steps to reproduce the behavior

  1. `$ docker compose version`

**Comment by nicks** [comment]
I don't think we do anything to prevent you from installing multiple versions of docker compose in WSL. do you know where the other one is coming from? e.g., what does `which docker-compose` show?

**Comment by CybotTM** [comment]
Hi @nicks ,

thanks for your reply

> I don't think we do anything to prevent you from installing multiple versions of docker compose in WSL. do you know where the other one is coming from?

It's the one provided by Docker Desktop.

> what does which docker-compose show?

see issue description:
```
$ which docker-compose
/usr/bin/docker-compose
$ ll /usr/bin/docker-compose
lrwxrwxrwx 1 root root 56 Aug 24 09:03 /usr/bin/docker-compose -> /mnt/wsl/docker-desktop/cli-tools/usr/bin/docker-compose*
```

**Comment by CybotTM** [comment]
Installing the latest update to Docker Desktop for Win 4.12.0 solved the issue for me:
```
sme@IOT-159:~$ docker compose version
Docker Compose version v2.10.2
sme@IOT-159:~$ docker-compose version
Docker Compose version v2.10.2
sme@IOT-159:~$ which docker-compose
/usr/bin/docker-compose
sme@IOT-159:~$ ll /usr/bin/docker-compose
lrwxrwxrwx 1 root root 56 Sep  6 13:29 /usr/bin/docker-compose -> /mnt/wsl/docker-desktop/cli-tools/usr/bin/docker-compose*
```

Dunno why the previous updates did not update the (docker-compose) binaries in `/mnt/wsl/docker-desktop/cli-tools/usr/bin/`.

**Comment by CybotTM** [comment]
The issue re-surfaced, I'm back at docker-compose 2.2.2 again

```
$ docker compose version
Docker Compose version v2.2.2
$ docker-compose version
Docker Compose version v2.2.2
$ which docker-compose
/usr/bin/docker-compose
$ l /usr/bin/docker-compose
/usr/bin/docker-compose@
$ ll /usr/bin/docker-compose
lrwxrwxrwx 1 root root 56 Oct 26 11:40 /usr/bin/docker-compose -> /mnt/wsl/docker-desktop/cli-tools/usr/bin/docker-compose*
$ ll /mnt/wsl/docker-desktop/cli-tools/usr/bin
total 155168
drwxr-xr-x 2 root root     2048 Aug 29 13:22 ./
drwxr-xr-x 6 root root     2048 Aug 29 13:21 ../
-rwxr-xr-x 1 root root 60513160 Jun  7 01:01 com.docker.cli*
-rwxr-xr-x 1 root root 56389632 Aug 29 13:21 docker*
-rwxr-xr-x 1 root root  6332416 Aug 29 13:21 docker-compose*
-rwxr-xr-x 1 root root 12737304 Aug 29 13:21 docker-compose-v1*
-rwxrwxr-x 1 root root  9709056 Aug 29 13:21 docker-credential-desktop.exe*
-rwxr-xr-x 1 root root 13205504 Aug 29 13:21 hub-tool*
$ uname -a
Linux Grosser 5.15.68.1-microsoft-standard-WSL2 #1 SMP Mon Sep 19 19:14:52 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
$ docker version
Client: Docker Engine - Community
 Cloud integration: v1.0.29
 Version:           20.10.17
 API version:       1.41
 Go version:        go1.17.11
 Git commit:        100c701
 Built:             Mon Jun  6 23:03:17 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Desktop
 Engine:
  Version:          20.10.17
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.11
  Git commit:       a89b842
  Built:            Mon Jun  6 23:01:23 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.8
  GitCommit:        9cd3357b7fd7218e4aec3eae239db1f68a5a6ec6
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by CybotTM** [comment]
but also:

![image](https://user-images.githubusercontent.com/326348/198008516-d3292761-76f0-46d4-8556-f74dffc4bd15.png)

**Comment by CybotTM** [comment]
```
$ mount
...
/dev/sdf on /mnt/wsl/docker-desktop-data/isocache type ext4 (rw,relatime,discard,errors=remount-ro,data=ordered)
none on /mnt/wsl/docker-desktop/shared-sockets/guest-services type tmpfs (rw,nosuid,nodev,mode=755)
none on /mnt/wsl/docker-desktop/shared-sockets/host-services type tmpfs (rw,nosuid,nodev,mode=755)
/dev/sdd on /mnt/wsl/docker-desktop/docker-desktop-user-distro type ext4 (rw,relatime,discard,errors=remount-ro,data=ordered)
/mnt/host/wsl/docker-desktop-data/isocache/entries/docker-wsl-cli.iso/81d0677385ac2178f12e63836ba0974911026c20e528d7db304c190ad97d57d5 on /mnt/wsl/docker-desktop/cli-tools type iso9660 (ro,relatime,nojoliet,check=s,map=n,blocksize=2048,iocharset=iso8859-1)
```
```
$ ll /mnt/host/wsl/docker-desktop-data/isocache/entries/docker-wsl-cli.iso/
ls: cannot access '/mnt/host/wsl/docker-desktop-data/isocache/entries/docker-wsl-cli.iso/': No such file or directory
```
```
$ ll /mnt/wsl/docker-desktop-data/isocache/entries/docker-wsl-cli.iso/
total 393600
drwxr-xr-x 2 root root      4096 Oct 26 12:27 ./
drwxr-xr-x 4 root root      4096 Oct 26 12:27 ../
-rw-r--r-- 1 root root 403038208 Oct 26 12:27 81d0677385ac2178f12e63836ba0974911026c20e528d7db304c190ad97d57d5
```

**Comment by CybotTM** [comment]
The mounted ISO holds the outdated binaries:

![image](https://user-images.githubusercontent.com/326348/198014058-678aa77c-29ff-46bf-ae6e-812aac35264b.png)

**Comment by CybotTM** [comment]
Solved for me in #13038
